### ISSUE TITLE
Stack is missing cabal's --extra-framework-dirs

### ISSUE THREAD

**Issue by TravisWhitaker** [commercialhaskell/stack#4022]
I'm working on a macOS-specific package that needs to link against frameworks. With cabal, I need to configure like this:

```bash
$ cabal configure --extra-framework-dirs=/System/Library/Frameworks/ ...
```

But stack seems to have no equivalent option. I'm sure it's possible to get around this with --ghc-options, but it'd be nice to be able to stick this in stack.yaml, just like extra-lib-dirs. I'd imagine that extra-framework-dirs works just like extra-lib-dirs and that this would be easy to add, but I don't know for sure.

**Comment by mgsloan** [comment]
Indeed, this is not currently supported.  Stack should really support arbitrary configure-options - it's been a longstanding omission - https://github.com/commercialhaskell/stack/issues/1438

Alternatively, support for this specific option could be added.

**Comment by karamellpelle** [comment]
@TravisWhitaker : What about adding the cabal option `extra-framework-dirs` to your _.cabal_ file?

`extra-framwork-dirs` was [introduced by Cabal 1.24.0.0](https://hackage.haskell.org/package/Cabal/changelog)?

**Comment by TravisWhitaker** [comment]
One can't assume what the value of `extra-framework-dirs` should be, especially if Nix is in use.
